Add Details to Contributor Guide About Forking/PRs (#59)
Azure/PyRIT: doc/contributing.md

    az login
    ```

> ## Set up for Contribution with Git

Reviewer: ```suggestion
## Set up for contribution with git
```

Author: Changed to be a lil more clear and proper heading caps

`Set Up: Contribute with Git`

> Before creating your first Pull Request, set up to contribute to PyRIT by following these steps:

Reviewer: ```suggestion
Before creating your first pull request, set up your fork to contribute to PyRIT by following these steps:
```

1. [Fork](https://github.com/Azure/PyRIT/fork) the repo from the main branch. By default, forks are named the same as their upstream repository. This will create a new repo called `GITHUB_USERNAME/PyRIT` (where `GITHUB_USERNAME` is a variable for your GitHub username).
2. Add this new repo locally wherever you cloned PyRIT
```
# to see existing remotes
git remote -v

# add your fork as a remote named `REMOTE_NAME`
git remote add REMOTE_NAME https://github.com/GITHUB_USERNAME/PyRIT.git
```

> To add your contribution to the repo, the flow typically looks like:

Reviewer: ```suggestion
To add your contribution to the repo, the flow typically looks as follows:
```

Author: updated

```

> # execute commands from wherever you cloned PyRIT locally

Reviewer: this is obvious from previous commands

Author: removed

git checkout main
git pull # pull from origin

> git checkout -b users/GITHUB_USERNAME/mybranch

Reviewer: I don't think we necessarily care what people call their branches, but I guess this is fine.

Author: i was including this as an example, but could just change to mybranch

Author: back to mybranch
